utils/events.py

```diff
@@ -135,3 +153,305 @@ def ensure_event_happens(self, target_type, target_id, event_type):
         if len(events) == 0:
             raise AssertionError(
                 'Event {}/{}/{} did not happen.'.format(event_type, target_type, target_id))
+
+
+class EventBuilder(object):
+    """
+    this class just simplifies "expected" event creation.
+
+    Usage:
+        builder = EventBuilder()
+        evt = builder.new_event(target_type='VmOrTemplate',
+                                target_name='my_lovely_vm',
+                                event_type='vm_create')
+    """
+    def __init__(self):
+        self._tool = EventTool(get_or_create_current_appliance())
+
+    def new_event(self, *attrs, **kwattrs):
+        event = Event(event_tool=self._tool)
+        for name, value in kwattrs.items():
+            event.add_attrs(EventAttr(**{name: value}))
+
+        for attr in attrs:
+            event.add_attrs(EventAttr(**attr))
+        return event
+
+
+class EventAttr(object):
+    """
+    contains one event attribute and the method for comparing it.
+    """
+    def __init__(self, attr_type=None, cmp_func=None, **attrs):
+        if len(attrs) > 1:
+            raise ValueError('event attribute can have only one key=value pair')
+
+        self.name, self.value = attrs.items()[-1]
+        self.type = attr_type or type(self.value)
+        self.cmp_func = cmp_func
```

Comment:
cmp_func was implemented for cases when we have to compare values using regexp, etc.
I'd implement __cmp__ otherwise.
If I used C++, I'd probably define custom operator. 
But in Python I didn't find better solution than implementing match() method.
I'd be happy if you adviced me something better.